### PR TITLE
update contributing guide

### PR DESCRIPTION
**Pull request by vankichi**
Signed-off-by: vankichi <kyukawa315@gmail.com>

### Description:

Update branch naming rule and add code blocking.

### Environment:

- Go Version: 1.15.1
- Docker Version: 19.03.8
- Kubernetes Version: 1.18.2
- NGT Version: 1.12.1

### Types of changes:

- [ ] Bug fix [type/bug]
- [ ] New feature [type/feature]
- [ ] Add tests [type/test]
- [ ] Security related changes [type/security]
- [x] Add documents [type/documentation]
- [ ] Refactoring [type/refactoring]
- [ ] Update dependencies [type/dependency]
- [ ] Update benchmarks and performances [type/bench]
- [ ] Update CI [type/ci]

### Changes to Core Features:

- [ ] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your core changes, as applicable?
- [ ] Have you successfully ran tests with your changes locally?

### Checklist:

- [x] I have read the [CONTRIBUTING](https://github.com/vdaas/vald/blob/master/CONTRIBUTING.md) document.
- [x] I have checked open [Pull Requests](https://github.com/vdaas/vald/pulls) for the similar feature or fixes?
- [x] I have added tests and benchmarks to cover my changes.
- [x] I have ensured all new and existing tests passed.
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -121,6 +123,6 @@ Name your branches with prefixes: `[type]/[area]/[description]`
 
 * `type` = feature, bug, refactoring, benchmark, security, documentation, dependencies, ci, ...
 * `area` (\*) = gateway, meta, manager-backup, manager-replication, ...
-* `description` = branch description. description must be hyphenated.
+* `description` = branch description. description must be hyphenated. please use [a-zA-Z0-9] and hyphen as characters, do not use any other characters.
```
Comment (hlts2) [inline CONTRIBUTING.md:126]:
```suggestion
* `description` = branch description. The description must be hyphenated. please use [a-zA-Z0-9] and hyphen as characters, do not use any other characters.
```

Comment (vankichi) [reply CONTRIBUTING.md:126]:
```suggestion
* `description` = branch description. The description must be hyphenated. Please use [a-zA-Z0-9] and hyphen as characters, do not use any other characters.
```

Comment (vankichi) [reply CONTRIBUTING.md:126]: @hlts2 How is it?

Comment (kpango) [reply CONTRIBUTING.md:126]: LGTM, but I have a idea.
IMO: using table style would be more user friendly

```html
<table>
<tr><td>type<td>type represents xxx<td>feature, bug, refactoring, benchmark, security, documentation, dependencies, ci, (XXX define every patter)
<tr><td>area (*)<td>this field represents area or context<td>gateway, meta, manager-backup, manager-replication, ... (XXX show more example)
<tr><td>description<td>summarized description of branch<td> branch description. description must be hyphenated. please use [a-zA-Z0-9] and hyphen as characters, do not use any other characters.
</table>
```

Comment (kevindiu) [reply CONTRIBUTING.md:126]: sorry but I want to confirm `description` means the branch name?

Comment (hlts2) [reply CONTRIBUTING.md:126]: > How is it?

it seems good to me!
and I also think that the table structure was easier to understand!

Comment (vankichi) [reply CONTRIBUTING.md:126]: @kevindiu 
It is almost right.
branch name is [type]/[area]/[description]

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -119,8 +121,10 @@ Your pull request is much more likely to be accepted if:
 
 Name your branches with prefixes: `[type]/[area]/[description]`
 
-* `type` = feature, bug, refactoring, benchmark, security, documentation, dependencies, ci, ...
-* `area` (\*) = gateway, meta, manager-backup, manager-replication, ...
-* `description` = branch description. description must be hyphenated.
+| Field       | Explanation                           | nameing rule      |
```
Comment (kevindiu) [inline CONTRIBUTING.md:124]:
```suggestion
| Field       | Explanation                           | Example      |
```

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -119,8 +121,10 @@ Your pull request is much more likely to be accepted if:
...
+| :---        | :---                                  | :---              |
+| type        | type represents the PR type           | feature, bug, refactoring, benchmark, security, documentation, dependencies, ci, ... |
```
Comment (kevindiu) [inline CONTRIBUTING.md:126]:
```suggestion
| type        | The PR type           | feature, bug, refactoring, benchmark, security, documentation, dependencies, ci, ... |
```

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -119,8 +121,10 @@ Your pull request is much more likely to be accepted if:
...
+| area        | this field represents area of context | gateway, meta, manager-backup, manager-replication, ... |
```
Comment (kevindiu) [inline CONTRIBUTING.md:127]:
```suggestion
| area        | The context of the PR | gateway, meta, manager-backup, manager-replication, ... |
```

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -119,8 +121,10 @@ Your pull request is much more likely to be accepted if:
...
+| description | summarized description of your branch | Branch description. The description must be hyphenated. Please use [a-zA-Z0-9] and hyphen as characters, do not use any other characters. |
```
Comment (kevindiu) [inline CONTRIBUTING.md:128]:
```suggestion
| description | Summarized description of your branch. Please use [a-zA-Z0-9] and hyphen characters for the description, and do not use any other characters. | this-is-branch-description  |
```

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -119,8 +121,10 @@ Your pull request is much more likely to be accepted if:
...
```
Comment (kevindiu) [inline CONTRIBUTING.md:129]: maybe it is better to add an example for branch name?

**Review by kevindiu — CHANGES_REQUESTED** [review]

File: CONTRIBUTING.md (hunk 2 above)
Comment (vankichi) [reply CONTRIBUTING.md:124]: I think this column is not example.

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -119,8 +121,12 @@ Your pull request is much more likely to be accepted if:
 
 Name your branches with prefixes: `[type]/[area]/[description]`
 
-* `type` = feature, bug, refactoring, benchmark, security, documentation, dependencies, ci, ...
-* `area` (\*) = gateway, meta, manager-backup, manager-replication, ...
-* `description` = branch description. description must be hyphenated.
+| Field | Explanation | Naming Rule |
+| :--- | :--- | :--- |
+| type | The PR type | feature, bug, refactoring, benchmark, security, documentation, dependencies, ci, ... |
```
Comment (kevindiu) [inline CONTRIBUTING.md:126]:
```suggestion
| type | The PR type | The type of the PR can be feature, bug, refactoring, benchmark, security, documentation, dependencies, ci, test, ... |
```

File: CONTRIBUTING.md [hunk 8]
```diff
@@ -119,8 +121,12 @@ Your pull request is much more likely to be accepted if:
...
+| area | Area of context | gateway, meta, manager-backup, manager-replication, ... |
```
Comment (kevindiu) [inline CONTRIBUTING.md:127]:
```suggestion
| area | Area of context | The context of the PR can be gateway, meta, manager-backup, manager-replication, ... |
```

**Review by hlts2 — APPROVED** [review]

**Review by kevindiu — DISMISSED** [review]
LGTM

**Pull request merged**